stdlib/messaging/nats/src/main/java/org/ballerinalang/nats/streaming/message/Ack.java

```diff
@@ -31,8 +31,14 @@
 
     public static Object externAck(ObjectValue message) {
         Message streamingMessage = (Message) message.getNativeData(Constants.NATS_STREAMING_MSG);
+        boolean manualAck = (Boolean) message.getNativeData(Constants.NATS_STREAMING_MANUAL_ACK);
         try {
-            streamingMessage.ack();
+            if (manualAck) {
+                streamingMessage.ack();
+            } else {
+                return BallerinaErrors.createError(Constants.NATS_ERROR_CODE, "error in trying to manually" +
+                        " acknowledge the message in auto ack mode.");
```

Comment:
Shall we make the error message like "Invalid operation, manual acknowledge is not supported in auto ACK mode."

Reply:
Agreed. Sounds better.